# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PContent.ipdl
# Commit: 095b3edec3c8
# Full Hash: 095b3edec3c8ba9318c71d0acc91e10ce57818dc
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2019-04-29 21:53:38
# Regressor Bug: 1514869
# File Overlap Count: 1
# Description:
#   Bug 1514869 - patch 2 - Adapt platform-font-list code to work with either the existing in-process font list or cross-process shared font list. r=jwatt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D22938
# ==============================================================================

diff -r 21ef00977ab6 -r 095b3edec3c8 dom/ipc/PContent.ipdl
--- a/dom/ipc/PContent.ipdl	Sat Apr 27 15:37:29 2019 +0000
+++ b/dom/ipc/PContent.ipdl	Sat Apr 27 15:37:58 2019 +0000
@@ -495,6 +495,19 @@
 
     async UpdateFontList(SystemFontListEntry[] fontList);
 
+    /**
+     * The shared font list has been reinitialized by the parent;
+     * child processes must discard and recreate their mappings to it.
+     */
+    async RebuildFontList();
+
+    /**
+     * The shared font list has been modified, potentially adding matches
+     * for src:local() names that were previously not known, so content
+     * may need to be reflowed.
+     */
+    async FontListChanged();
+
     async UpdateAppLocales(nsCString[] appLocales);
     async UpdateRequestedLocales(nsCString[] requestedLocales);
 